backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/attribute/impl/HUUniqueAttributesService.java

```diff
@@ -108,6 +109,19 @@ public void validateHU(@NonNull final HuId huId)
 				.forEach(this::validateHUUniqueAttribute);
 	}
 
+	public void validateAttribute(@NonNull final I_M_Attribute attribute)
+	{
+		Services.get(IQueryBL.class).createQueryBuilder(I_M_HU_Attribute.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_M_HU_Attribute.COLUMNNAME_M_Attribute_ID, attribute.getM_Attribute_ID())
+				.addEqualsFilter(I_M_HU_Attribute.COLUMNNAME_Value, attribute.getValue())
+				.create()
+				.list(I_M_HU_Attribute.class)
+				.stream()
+				.filter(I_M_HU_Attribute::isUnique)
```

Comment:
Q: why not using using `addEqualsFilter(I_M_HU_Attribute.COLUMNNAME)IsUnique, true)` ?
(so you would apply the filtering on database side)